src/blueapi/service/runner.py

```diff
@@ -106,3 +111,24 @@ def state(self) -> EnvironmentResponse:
 class InvalidRunnerStateError(Exception):
     def __init__(self, message):
         super().__init__(message)
+
+
+class RpcError(Exception): ...
+
+
+def _rpc(
+    module_name: str, function_name: str, *args: P.args, **kwargs: P.kwargs
+) -> Any:
+    mod = import_module(module_name)
+    func: Callable[P, T] = _validate_function(
+        mod.__dict__.get(function_name, None), function_name
+    )
+    return func(*args, **kwargs)
```

Comment:
@DiamondJoseph as discussed...

I don't think `P` is useful here, it doesn't bind the function and pyright complains about it. Because this is RPC the parameter and return types are not knowable at "compile" time, which means you will need to write runtime code to verify that your type hints reflect reality. I suggest allowing `args` and `kwargs` to be `Any` in this instance because verifying their type is hard and not _that_ useful:

```suggestion
def _rpc(
    module_name: str, function_name: str, *args: Any, **kwargs: Any
) -> Any:
    mod = import_module(module_name)
    func: Callable[P, T] = _validate_function(
        mod.__dict__.get(function_name, None), function_name
    )
    return func(*args, **kwargs)
```

What is easier, and probably more useful, is to validate the _return_ type at runtime:

```suggestion
def _rpc(
    module_name: str, function_name: str, expected_return_type: T, *args: P.args, **kwargs: P.kwargs
) -> T:
    mod = import_module(module_name)
    func: Callable[P, T] = _validate_function(
        mod.__dict__.get(function_name, None), function_name
    )
    value = func(*args, **kwargs)
    if isinstance(value, expected_return_type):
        return value
    else:
        raise TypeError()
```